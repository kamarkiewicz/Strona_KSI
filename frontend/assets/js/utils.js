
// Access image field from backend
function getImage (image) {
  if ((image || {}).path) {
    // path may be relative URL
    let isAbsolute = /^(?:[a-z]+:)?\/\//i.test(image.path)
    image.path = (isAbsolute ? '' : process.env.API_URL) + image.path
  }
  return image
}

export { getImage };
