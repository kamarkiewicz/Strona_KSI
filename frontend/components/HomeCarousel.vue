<template>
  <div class="py-5 text-white bg-secondary">
    <div class="container">
      <div class="row">
        <div class="align-self-center p-5 col-md-6">
          <h1 class="mb-4">{{ title }}</h1>
          <p class="mb-5">{{ desc }}</p>
          <a class="btn btn-lg btn-outline-light" href="#">{{ linkDesc }}</a>
        </div>
        <div class="col-md-6 p-0">
          <b-carousel id="carousel1"
                      style="text-shadow: 1px 1px 2px #333;"
                      controls
                      indicators
                      background="#ababab"
                      :interval="4000"
                      img-width="500"
                      img-height="500"
                      v-model="slide"
                      @sliding-start="onSlideStart"
                      @sliding-end="onSlideEnd"
          >
            <b-carousel-slide v-for="slide in slides" :key="slide.id"
                              :caption="slide.caption"
                              :text="slide.text"
                              :img-src="slide.imgSrc"
            ></b-carousel-slide>
          </b-carousel>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  data () {
    return {
      title: 'Zobacz na własne oczy!',
      desc: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
      linkDesc: 'Przejdź do galerii',
      slides: [
        { id: 1,
          caption: 'First slide',
          text: 'Nulla vitae elit libero, a pharetra augue mollis interdum.',
          imgSrc: 'https://picsum.photos/500/500?image=1033'
        },
        { id: 2,
          caption: 'Second slide',
          text: 'Nulla vitae elit libero, a pharetra augue mollis interdum.',
          imgSrc: 'https://picsum.photos/500/500?image=1060'
        },
        { id: 3,
          caption: 'Third slide',
          text: 'Nulla vitae elit libero, a pharetra augue mollis interdum.',
          imgSrc: 'https://picsum.photos/500/500?image=1059'
        }
      ],

      slide: 0,
      sliding: null
    }
  },
  methods: {
    onSlideStart (slide) {
      this.sliding = true
    },
    onSlideEnd (slide) {
      this.sliding = false
    }
  }
}
</script>

<style scoped>
</style>
