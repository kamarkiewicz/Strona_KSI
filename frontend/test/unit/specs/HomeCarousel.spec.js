import Vue from 'vue'
import BootstrapVue from 'bootstrap-vue'
import { mount } from '@vue/test-utils'
import HomeCarousel from '~/components/HomeCarousel.vue'
import { Image } from '~/assets/js/utils'

Vue.use(BootstrapVue)
Vue.config.ignoredElements = [
  'nuxt-link'
];

describe('HomeCarousel', () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(HomeCarousel, {
      mocks: {
        $t: (label) => { return label },
        localePath: (args) => { return JSON.stringify(args) },
      },
      propsData: {
        slides: [
          {
            caption: 'First slide',
            description: 'Nulla vitae elit libero, a pharetra augue mollis interdum.',
            image: new Image('http://via.placeholder.com/500x500'),
          },
          {
            caption: 'Second slide',
            description: 'Nulla vitae elit libero, a pharetra augue mollis interdum.',
            image: new Image('http://via.placeholder.com/500x500'),
          },
          {
            caption: 'Third slide',
            description: 'Nulla vitae elit libero, a pharetra augue mollis interdum.',
            image: new Image('http://via.placeholder.com/500x500'),
          },
        ],
      },
    })
  })

  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
