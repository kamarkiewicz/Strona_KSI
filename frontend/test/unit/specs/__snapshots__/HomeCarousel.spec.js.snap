// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeCarousel renders correctly 1`] = `
<div class="home-carousel pb-5 text-white bg-secondary">
  <div class="container">
    <div class="row">
      <div class="align-self-center p-5 col-md-6">
        <h1 class="mb-4">homecarousel.title</h1>
        <p class="mb-5">homecarousel.description</p>
        <nuxt-link to="&quot;album&quot;" class="btn btn-lg btn-outline-light">homecarousel.gotoAlbum</nuxt-link>
      </div>
      <div class="col-md-6 p-0">
        <b-carousel id="carousel1" controls="" indicators="" background="#ababab" interval="4000" img-width="500" img-height="500" style="text-shadow: 1px 1px 2px #333;">
          <b-carousel-slide caption="First slide" text="Nulla vitae elit libero, a pharetra augue mollis interdum." img-src="http://via.placeholder.com/500x500" img-alt="Featured image"></b-carousel-slide>
          <b-carousel-slide caption="Second slide" text="Nulla vitae elit libero, a pharetra augue mollis interdum." img-src="http://via.placeholder.com/500x500" img-alt="Featured image"></b-carousel-slide>
          <b-carousel-slide caption="Third slide" text="Nulla vitae elit libero, a pharetra augue mollis interdum." img-src="http://via.placeholder.com/500x500" img-alt="Featured image"></b-carousel-slide>
        </b-carousel>
      </div>
    </div>
  </div>
</div>
`;
